code/extensions/che-activity-tracker/src/activity-tracker-service.ts

```diff
@@ -67,17 +72,82 @@ export class ActivityTrackerService {
 		}
 	}
 
-	private sendRequest(
-		attemptsLeft: number = ActivityTrackerService.RETRY_COUNT,
-	): void {
+	private async sendRequest(
+		attemptsLeft: number = ActivityTrackerService.RETRY_COUNT
+	): Promise<void> {
 		try {
-			this.workspaceService.updateWorkspaceActivity();
+			await this.workspaceService.updateWorkspaceActivity();
 		} catch (error) {
 			if (attemptsLeft > 0) {
-			  setTimeout(this.sendRequest, ActivityTrackerService.RETRY_REQUEST_PERIOD_MS, --attemptsLeft);
+				await new Promise((resolve) => setTimeout(resolve, ActivityTrackerService.RETRY_REQUEST_PERIOD_MS));
+				await this.sendRequest(--attemptsLeft);
 			} else {
-			  console.error('Activity tracker: Failed to ping che-machine-exec: ', error.message);
+				this.channel.appendLine('Activity tracker: Failed to ping che-machine-exec: ' + error.message);
+				if (!this.errorDisplayed) {
+					this.errorDisplayed = true;
+					this.showErrorMessage();
```

Comment:
I suppose we have to set `this.errorDisplayed` back to `false` when the popup is closed.
```suggestion
					await this.showErrorMessage();
					this.errorDisplayed = false;
```

Reply:
@vitaliy-guliy thank you for the review, I didn't set it back to false so that the error message appears only once instead of every minute, WDYT?